libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -959,7 +993,15 @@ string YulUtilFunctions::storageArrayIndexAccessFunction(ArrayType const& _type)
 
 				let data := <dataAreaFunc>(array)
 				<?multipleItemsPerSlot>
-
+					<?isBytesArray>
+						let itemsPerSlot := div(0x20, <storageBytes>)
+						slot := add(data, div(index, itemsPerSlot))
+						offset := mod(index, itemsPerSlot)
+						if lt(<arrayLen>(array), 0x20) {
+							slot := array
+							offset := add(offset, 1)
```

Comment:
Is this needed?

Reply:
I can remove this part, but will need to change `updateByteSliceFunctionDynamic` to shift left for `31 - shiftBytes`, instead for `shiftBytes` like now

Reply:
Ah right, I confused big- and little-endian here.